src/Pate/Proof/Instances.hs

```diff
@@ -225,17 +226,17 @@ instance forall sym arch tp. (PA.ValidArch arch, PSo.ValidSym sym) => PP.Pretty
 
 instance forall sym arch.
   (PA.ValidArch arch, PSo.ValidSym sym) =>
-  PP.Pretty (PF.EquivalenceDomain sym arch) where
+  PP.Pretty (PED.EquivalenceDomain sym arch) where
   pretty prf = PP.vsep
     [ "Registers:"
     , PP.indent 4 $ prettyRegs
-    , "Stack Memory:" <+> ppPolarity (PEM.memDomainPolarity $ PF.eqDomainStackMemory prf)
-    , PP.indent 4 $ prettyMem (PF.eqDomainStackMemory prf)
-    , "Global Memory:" <+> ppPolarity (PEM.memDomainPolarity $ PF.eqDomainGlobalMemory prf)
-    , PP.indent 4 $ prettyMem (PF.eqDomainGlobalMemory prf)
+    , "Stack Memory:" <+> ppPolarity (PEM.memDomainPolarity $ PED.eqDomainStackMemory prf)
+    , PP.indent 4 $ prettyMem (PED.eqDomainStackMemory prf)
+    , "Global Memory:" <+> ppPolarity (PEM.memDomainPolarity $ PED.eqDomainGlobalMemory prf)
+    , PP.indent 4 $ prettyMem (PED.eqDomainGlobalMemory prf)
     ]
     where
-      prettyRegs = PP.vsep (map ppReg (collapseRegState (Proxy @sym) (PF.eqDomainRegisters prf)))
+      prettyRegs = PP.vsep (map ppReg (collapseRegState (Proxy @sym) (PED.eqDomainRegisters prf)))
 
       ppReg :: (Some (MM.ArchReg arch), W4.Pred sym) -> PP.Doc a
       ppReg (Some reg, p) = case W4.asConstantPred p of
```

Comment:
Could this function be defined next to the register domain to preserve some abstraction?

Reply:
Do you mean 'collapseRegState' in particular, or the pretty printing logic? I've wrapped RegisterDomain in a newtype now, so converting it to a list of register-predicate pairs now at least has to go through that interface.

Reply:
I was thinking about the pretty printing logic for registers in general. If there was just a pretty printer (i.e., this pretty printer) for the domain registers (defined with that newtype), this would just be `ppDomainRegisters` and you don't even need to expose the conversion to a list.

Reply:
Sure I can shuffle that into the RegisterDomain module (probably more examples in that module now that can be moved closer to their respective types). This logic still ends up somewhat duplicated in the Render.Proof module though, since it doesn't want to just produce a PP.Doc.

Reply:
ah that is fair. The duplication in the UI is truly unfortunate and I wish I had a better answer there